docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+# StdoutExporter Design
+
+In strongly typed languages typically there will be 2 separate `Exporter` interfaces, one that accepts spans from a tracer (SpanExporter) and one that accepts metrics (MetricsExporter)
+
+The exporter SHOULD be called with a checkpoint of finished (possibly dimensionally reduced) export records. Most configuration decisions have been made before the exporter is invoked, including which instruments are enabled, which concrete aggregator types to use, and which dimensions to aggregate by.
+
+## Use Cases
+
+Monitoring and alerting systems commonly use the data provided through metric events or tracers, after applying various aggregations and converting into various exposition format. After getting the data, the systems need to be able to see the data. The StdoutExporter will be used here to print data through Stdout, this is seen as a simple exporter where the user doesn’t have the burden of implementing or setting up a protocol dependent exporter.
+
+The StdoutExporter will also be used as a debugging tool for the Metrics API/SDK and Tracing API/SDK which are currently work in progress projects. This exporter will allow contributors to easily diagnose problems when working on the project.
+
+## Push vs Pull Exporter
+
+There are two different versions of exporters: Push and Pull. A Push Exporter pushes the data outwards towards a system, in the case of the StdoutExporter it prints data into Stdout. A Pull Exporter exposes data to some endpoint for another system to grab the data. 
+
+The StdoutExporter will only be implementing a Push Exporter framework.
+
+## Design Tenets
+
+* Reliability
+    * The Exporter should be reliable; data exported should always be accounted for. The data will either all be successfully exported to the destination server, or in the case of failure, the data is dropped. `Export` will always return failure or success to notify the user of the result.
+    * Thread Safety
+        * Export() should never be called concurrently. This is handled by the controller in the StdoutMetricsExporter.
+* Scalability
+    * The Exporter must be able to operate on sizeable systems with predictable overhead growth.  A key requirement of this is that the library does not consume unbounded memory resource. 
+* Security
+    * While authorization is required in protocol-specific exporters for third-party platforms, the StdoutExporter has no authentication since there is no third-party to communicate with.
```

Comment:
Probably consider "Stdout exporter should only be used for development and testing purpose, where security and privacy is less a concern as it doesn't communicate to external systems".

Reply:
This is definitely a better and more eloquent way of saying it, will change it